### PR TITLE
Fixed popovers to not error when using message stream

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
Added null check to fix the issue. However, this brought up an issue with our demo - the current demo for message stream wasn't checking on the popover's open state before sending an `Open` message, sending the popover an unnecessary message. I've updated the demo to add logic that checks the open state before sending.

#94

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-popovers?_sp.spa=pr-popover-animation-null-check](https://developer.blackbaud.com/skyux-popovers?_sp.spa=pr-popover-animation-null-check)

File: src/app/public/plugin-resources/code-examples/popover/programmatic/popover-demo.component.ts [hunk 1]
```diff
@@ -19,8 +19,16 @@ export class PopoverDemoComponent {
 
   public popoverController = new Subject<SkyPopoverMessage>();
 
+  private popoverOpen: boolean = false;
+
+  public onPopoverOpen(isOpen: boolean): void {
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/plugin-resources/code-examples/popover/programmatic/popover-demo.component.ts:24]: Seems a little odd that we are calling this `onPopoverOpen` but also using it on close. Could we do something like `onPopoverStateChange` or similar?

**Review by Blackbaud-TrevorBurch — COMMENTED** [review]
Just one question

File: src/app/public/plugin-resources/code-examples/popover/programmatic/popover-demo.component.ts (hunk 1 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/plugin-resources/code-examples/popover/programmatic/popover-demo.component.ts]: Yep good call. Done.

**Review by Blackbaud-TrevorBurch — DISMISSED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-popovers?_sp.spa=pr-popover-animation-null-check](https://developer.blackbaud.com/skyux-popovers?_sp.spa=pr-popover-animation-null-check)

**Review by Blackbaud-TrevorBurch — APPROVED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-popovers?_sp.spa=pr-popover-animation-null-check](https://developer.blackbaud.com/skyux-popovers?_sp.spa=pr-popover-animation-null-check)

**Pull request merged**